### PR TITLE
Update ScalaFileUpload with tips for CSRF

### PR DESCRIPTION
**Pull request by bwbecker**
This new paragraph would have saved me several hours of debugging.  Having a "NoTokenInBody" error that depends on the order of fields in a form is not obvious!

I don't work with the Java version of Play, so have not updated that documentation.

I don't think I changed the part "The above configuration..."  Not sure why it's showing in the diff.

# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [ ] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #xxxx

## Purpose

What does this PR do?

## Background Context

Why did you take this approach?

## References

Are there any relevant issues / PRs / mailing lists discussions?

### CODE REVIEW

File: documentation/manual/working/scalaGuide/main/upload/ScalaFileUpload.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Start by writing an HTML form:
 
 @[file-upload-form](code/scalaguide/templates/views/uploadForm.scala.html)
 
+Add a CSRF token to the form, unless you have the CSRF filter disabled. The CSRF filter checks the multi-part form in the order the fields are listed, so put the CSRF token before the file input field. This improves efficiency and avoids a token-not-found error if the file size exceeds `play.filters.csrf.body.bufferSize`.
```
Comment (wsargent) [inline documentation/manual/working/scalaGuide/main/upload/ScalaFileUpload.md]: Please add a link in documentation to the CSRFFilter

**Comment by bwbecker** [comment]
@wsargent I've added a link to the CSRF documentation.  There are now three commits;  I think I remember that you like everything squashed into just one commit.  Should I squash and do a new pull request?  Relatively new to this and need guidance...

**Comment by gmethvin** [comment]
@bwbecker You can squash yourself and force push

**Comment by gmethvin** [comment]
Also we should update the Java documentation as well.

**Comment by bwbecker** [comment]
Hey guys, I still intend to squash the commits and take a stab at the Java documentation.  The start of the school year is always really busy for me; I'll need a couple more days to get to it.

**Comment by lightbend-cla-validator** [comment]
Hi @mergify[bot],

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by lightbend-cla-validator** [comment]
Hi @mergify[bot],

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Pull request merged**